src/main/java/io/jenkins/plugins/coverage/threshold/Threshold.java

```diff
@@ -1,41 +1,76 @@
 package io.jenkins.plugins.coverage.threshold;
 
+import hudson.Extension;
 import hudson.ExtensionPoint;
 import hudson.model.Describable;
 import hudson.model.Descriptor;
 import io.jenkins.plugins.coverage.targets.CoverageMetric;
 import jenkins.model.Jenkins;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.util.Objects;
 
 public class Threshold implements ExtensionPoint, Describable<Threshold> {
-    private final CoverageMetric threshTarget;
 
-    private final float healthyThresh;
-    private final float unstableThresh;
-    private final float unhealthyThresh;
+    private final CoverageMetric thresholdTarget;
+
+    // mark build as unstable when coverage is less than this
+    private float unstableThreshold = 0.0f;
+
+    // used for calculate healthy scores
+    private float unhealthyThreshold = 0.0f;
+
+    private boolean failUnhealthy = false;
 
     @DataBoundConstructor
-    public Threshold(CoverageMetric threshTarget, float healthyThresh, float unstableThresh, float unhealthyThresh) {
-        this.threshTarget = threshTarget;
-        this.healthyThresh = healthyThresh;
-        this.unstableThresh = unstableThresh;
-        this.unhealthyThresh = unhealthyThresh;
+    public Threshold(CoverageMetric thresholdTarget) {
+        this.thresholdTarget = thresholdTarget;
+    }
+
+
+    public CoverageMetric getThresholdTarget() {
+        return thresholdTarget;
+    }
+
+    public float getUnstableThreshold() {
+        return unstableThreshold;
+    }
+
+    @DataBoundSetter
+    public void setUnstableThreshold(float unstableThreshold) {
+        this.unstableThreshold = unstableThreshold;
+    }
+
+    public float getUnhealthyThreshold() {
+        return unhealthyThreshold;
+    }
+
+    @DataBoundSetter
+    public void setUnhealthyThreshold(float unhealthyThreshold) {
+        this.unhealthyThreshold = unhealthyThreshold;
     }
 
-    public CoverageMetric getThreshTarget() {
-        return threshTarget;
+    public boolean isFailUnhealthy() {
+        return failUnhealthy;
     }
 
-    public float getHealthyThresh() {
-        return healthyThresh;
+    @DataBoundSetter
+    public void setFailUnhealthy(boolean failUnhealthy) {
+        this.failUnhealthy = failUnhealthy;
     }
 
-    public float getUnstableThresh() {
-        return unstableThresh;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Threshold threshold = (Threshold) o;
+        return getThresholdTarget() == threshold.getThresholdTarget();
```

Comment:
Since you're using `Objects.hash` below for the hashcode it would be best to use `Objects.equal` in here instead of using the `==` operator.